### ISSUE TITLE
Gas limit is ignored.

### ISSUE THREAD

**Issue by MicahZoltu** [ethereum/remix-ide#1352]
In the run tab of remix I am prompted to set the gas limit for my transactions.  However, setting it doesn't appear to actually work and instead estimated gas is used.  This is problematic when gas estimation doesn't work.  For example, I have a transaction that has variable gas costs based on when it is mined, so I need to explicitly specify gas and not use estimates.

Gas limit setting:
![image](https://user-images.githubusercontent.com/886059/41139908-42e83366-6b1d-11e8-966d-b36ad6dec982.png)

Actual gas provided:
![image](https://user-images.githubusercontent.com/886059/41139921-4be240f6-6b1d-11e8-9323-12687be0db4f.png)

This is when using `Web3 Provider`.

**Comment by yann300** [comment]
yeah gas limit here is the max amount of gas... there should be a different semantic here

**Comment by cleanunicorn** [comment]
Yes, this is confusing. Could the gas limit specified there be used when making transactions? 

I see there is a gas estimation request done before and the result from there is used. Sometimes that is incorrect and even if we set a really high gas limit the transaction fails.

**Comment by cleanunicorn** [comment]
@yann300 I'm not sure if this is enough or if I break some functionality
https://github.com/ethereum/remix/pull/1092

**Comment by labanskoller** [comment]
I hit this problem while playing the 35C3 Junior CTF. I think it's a bug too, but I'm a newbie to Ethereum and Remix. This section of my challenge write-up describes my problem: https://labanskoller.se/blog/2019/01/07/solution-to-35c3-junior-ctf-challenge-entrance/#running-out-of-gas-and-problems-changing-the-limit
I've not tried the proposed fix in the linked pull request.
